### PR TITLE
W-16847560: use the same json schema version mapping in raml and oas

### PR DESCRIPTION
**Pull request by arielmirra**

### CODE REVIEW

File: amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala [hunk 1]
```diff
@@ -13,7 +14,7 @@ trait JsonSchemaInference {
     parseSchemaEntry(ast) match {
       case Some(node) =>
         tryParserVersion(node)(errorHandler) match {
-          case Some(version) => getVersionFor(version).getOrElse(defaultSchemaVersion)
+          case Some(version) => JsonSchemaEntry(version).getOrElse(defaultSchemaVersion)
```
Comment (looseale) [inline amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala]: The parent trait is calling the object that implement it. This is weird, I don't know why this is compiling...

File: amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala [hunk 2]
```diff
@@ -28,31 +29,15 @@ trait JsonSchemaInference {
 
   private def tryParserVersion(node: YNode)(errorHandler: AMFErrorHandler): Option[String] = {
     node.value match {
-      case scalar: YScalar => Some(adaptInput(scalar.text))
+      case scalar: YScalar => Some(scalar.text)
```
Comment (looseale) [inline amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala:48]: Don't remove methods

File: amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/oas/parser/InlineOasTypeParser.scala [hunk 3]
```diff
@@ -1060,7 +1060,6 @@ case class InlineOasTypeParser(
   }
 
   abstract class ShapeParser(implicit ctx: ShapeParserContext) {
```
Comment (looseale) [inline amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/oas/parser/InlineOasTypeParser.scala]: Don't commit files without code changes

**Review by looseale — CHANGES_REQUESTED** [review]

File: amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala (hunk 1 above)
Comment (arielmirra) [reply amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala]: done, moved logic to parent

File: amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala (hunk 2 above)
Comment (arielmirra) [reply amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/ref/JsonSchemaInference.scala:48]: it's not needed, the same method exists in parent

File: amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/oas/parser/InlineOasTypeParser.scala (hunk 3 above)
Comment (arielmirra) [reply amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/oas/parser/InlineOasTypeParser.scala]: removed

**Review by looseale — APPROVED** [review]

**Pull request merged**